src/librustc_codegen_llvm/common.rs

```diff
@@ -259,10 +259,16 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                 let base_addr = match alloc_kind {
                     Some(GlobalAlloc::Memory(alloc)) => {
                         let init = const_alloc_to_llvm(self, alloc);
+                        let formatted_id = if self.sess().fewer_names() {
```

Comment:
It seems that `static_addr_of{,_mut}` already performs this check:

https://github.com/rust-lang/rust/blob/ebbb2bf37aedaaa64dfaa52ba337ca6efb6b9093/src/librustc_codegen_llvm/consts.rs#L176

Reply:
How do we handle this path through `static_address_of`?

https://github.com/rust-lang/rust/blob/ebbb2bf37aedaaa64dfaa52ba337ca6efb6b9093/src/librustc_codegen_llvm/consts.rs#L322-L332

move the check into the `else` block in the changes here?

```rust
else {
    self.static_addr_of(init, alloc.align, formatted_id)
}
```

Reply:
Insertions into `const_globals` are below in the same function, so they should always originate from a call of `static_addr_of_mut`.

Reply:
As I understand it, I am ok to remove the check entirely from `src/librustc_codegen_llvm/common.rs` then.  I will push this change

Reply:
Done https://github.com/rust-lang/rust/pull/69155/commits/9a143854dc0899fe00dc36115235716acbccc79e